src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/mapper/HaFlowMapperTest.java

```diff
@@ -98,28 +106,31 @@ public void haFlowRequestMappingTest() {
         assertEquals(request.isStrictBandwidth(), result.isStrictBandwidth());
         assertEquals(request.getDescription(), result.getDescription());
         assertEquals(request.isAllocateProtectedPath(), result.isAllocateProtectedPath());
-        assertSharedEndpoint(request.getSharedEndpoint(), result.getSharedEndpoint());
+        assertEquals(request.getSharedEndpoint().getSwitchId(), result.getSharedSwitchId());
+        assertEquals(request.getSharedEndpoint().getPortNumber().intValue(), result.getSharedPort());
+        assertEquals(request.getSharedEndpoint().getOuterVlanId(), result.getSharedOuterVlan());
+        assertEquals(request.getSharedEndpoint().getInnerVlanId(), result.getSharedInnerVlan());
         // subflows must be mapped separately
         assertEquals(0, result.getSubFlows().size());
     }
 
     @Test
     public void getResponseTest() {
         HaFlow haFlow = new HaFlow(
```

Comment:
I suggest to use a builder here. It's difficult to track things like vlan and inner vlan are on right places.

Reply:
I'm not really like builders in tests because when you will add a new field into object you will have add it into tests because of compilation error. But in builder case it will just set it as "null"

Reply:
There are ways to combine this check with using a builder. For example, assertj has an [assertion](https://assertj.github.io/doc/#assertj-core-3.24.0-hasNoNullFields) for nulls in an object. If somebody adds a new field the old assertion starts fail and then somebody needs either to add ignoring that field, or add that parameter to a builder (update the test). 
This is just a comment, maybe it's worth to think about this kind of options next time.